File: testing/src/main/java/com/load/DTO/LoadTestConfigRequest.java
package com.load.DTO;

import java.time.LocalDateTime;



import lombok.Getter;
import lombok.Setter;

@Getter@Setter


public class LoadTestConfigRequest {
    private String fileName;
    private String targetUrl;
    private int numUsers;
    private int rampUpPeriod; // in seconds
    private int testDuration; // in seconds
    private LocalDateTime scheduledExecutionTime;
    
}

File: tehelka_testing_app/lib/load_test_config_request.dart
class LoadTestConfigRequest {
  final String testName;
  final String targetUrl;
  final int numUsers;
  final int rampUpPeriod;
  final int testDuration;
  final String crudType ;
  final DateTime scheduledExecutionTime; 

  LoadTestConfigRequest({
    required this.testName,
    required this.targetUrl,
    required this.numUsers,
    required this.rampUpPeriod,
    required this.testDuration,
    required this.scheduledExecutionTime,
    required this.crudType,
  });

  Map<String, dynamic> toJson() => {
        'testName': testName,
        'targetUrl': targetUrl,
        'numUsers': numUsers,
        'rampUpPeriod': rampUpPeriod,
        'testDuration': testDuration,
        'scheduledExecutionTime': scheduledExecutionTime,
        'crudType': crudType,
      };
}
